### PR TITLE
C++ unit tests for LLMNodeRunner, LLMTaskHandlerRunner and LLMTask

### PR DESCRIPTION
**Pull request by efajardo-nv**
## Description
- C++ unit tests for LLMNodeRunner, LLMTaskHandlerRunner and LLMTask
- Fix check for duplicate input names in LLMNodeRunner

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: morpheus/_lib/tests/llm/test_llm_node_runner.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+{
+    llm::LLMNode node;
+
+    auto runner_1 = node.add_node("Root1", {}, make_nr_dummy_node());
+    auto runner_2 = node.add_node("Root2", {{"/Root1"}}, make_nr_single_input_node(), true);
+
+    auto context = std::make_shared<llm::LLMContext>(llm::LLMTask{}, nullptr);
+
+    coroutines::sync_wait(runner_1->execute(context));
+    coroutines::sync_wait(runner_2->execute(context));
```
Comment (cwharris) [inline morpheus/_lib/tests/llm/test_llm_node_runner.cpp:115]: Important: run the executions in parallel.
```suggestion
    coroutines::sync_wait(
        coroutines::when_all(
            runner_1->execute(context),
            runner_2->execute(context)
        )
    );
```

File: morpheus/_lib/tests/llm/test_llm_node_runner.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+TEST_F(TestLLMNodeRunner, InvalidExternalNode)
+{
+    auto node   = make_nr_single_input_node();
+    auto inputs = llm::input_mappings_t{{"ext/input0", "arg0"}};
+    ASSERT_THROW(llm::LLMNodeRunner("runner", inputs, node), std::invalid_argument);
+}
+
+TEST_F(TestLLMNodeRunner, InvalidInputForNode)
```
Comment (cwharris) [inline morpheus/_lib/tests/llm/test_llm_node_runner.cpp:95]: Important: split this in to two tests.

File: morpheus/_lib/tests/llm/test_llm_node_runner.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+TEST_F(TestLLMNodeRunner, ParentInput)
+{
+    auto node   = make_nr_single_input_node();
+    auto inputs = llm::input_mappings_t{{"input0", "arg0"}};
+    llm::LLMNodeRunner runner{"runner", inputs, node};
+    ASSERT_EQ(runner.parent_input_names(), std::vector<std::string>{"input0"});
+    ASSERT_EQ(runner.sibling_input_names(), std::vector<std::string>{});
+}
+
+TEST_F(TestLLMNodeRunner, SiblingInput)
```
Comment (cwharris) [inline morpheus/_lib/tests/llm/test_llm_node_runner.cpp:66]: Important: add a test that uses a combination of parent and sibling inputs.

Comment (cwharris) [reply morpheus/_lib/tests/llm/test_llm_node_runner.cpp:66]: Important: split this in to 2 tests.

File: morpheus/_lib/src/llm/llm_node_runner.cpp [hunk 4]
```diff
@@ -60,9 +60,9 @@ LLMNodeRunner::LLMNodeRunner(std::string name, input_mappings_t inputs, std::sha
     {
         const auto& external_name = inp.external_name;
 
-        if (specified_input_names_set.contains(inp.external_name))
+        if (specified_input_names_set.contains(inp.internal_name))
         {
-            throw std::runtime_error("Input '" + inp.external_name + "' is specified more than once");
+            throw std::runtime_error("Input '" + inp.internal_name + "' is specified more than once");
```
Comment (cwharris) [inline morpheus/_lib/src/llm/llm_node_runner.cpp:65]: Question: we're sure this is a bug with the existing code?

File: morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        co_return results;
+    }
+
+  private:
+    std::vector<std::string> m_input_names;
+};
+
+TEST_CLASS(LLMTaskHandlerRunner);
+
+TEST_F(TestLLMTaskHandlerRunner, TryHandle)
```
Comment (cwharris) [inline morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp:87]: Question: what are we testing here? it appears the code paths we hit are all part of the test handler, which only exists to serve the test.

File: morpheus/_lib/src/llm/llm_node_runner.cpp (hunk 4 above)
Comment (efajardo-nv) [reply morpheus/_lib/src/llm/llm_node_runner.cpp:65]: Pretty sure. My `DuplicateInput` test wasn't throwing error before change. `specified_input_names_set` is not storing `internal_name` not `external_name` in line 104:
```
specified_input_names_set.insert(inp.internal_name);
```
Change doesn't break the existing unit tests.

File: morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp (hunk 5 above)
Comment (efajardo-nv) [reply morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp:87]: Thank you. I'm not even testing `LLMTaskHandlerRunner` here. I'll remove and use the test class only to test the non-virtual function `input_names()`.

Comment (efajardo-nv) [reply morpheus/_lib/tests/llm/test_llm_task_handler_runner.cpp]: code paths now go though tested class

**Review by cwharris — APPROVED** [review]

**Review by drobison00 — APPROVED** [review]

**Pull request merged**